beacon_node/network/src/sync/block_lookups/mod.rs

```diff
@@ -812,7 +824,55 @@ impl<T: BeaconChainTypes> BlockLookups<T> {
                 ))
             }
         } else {
-            Ok(stuck_lookup)
+            Ok(lookup)
+        }
+    }
+
+    /// Adds peers to a lookup and its ancestors recursively. Peers are added to the child first for
+    /// simplicity. We may choose to only add peers to the oldest ancestor. Child lookups should not
+    /// be able to make progress, but we add peers to them anyway for completeness.
+    /// Note: Takes a `lookup_id` as argument to allow recursion on mutable lookups, without having
+    /// to duplicate the code to add peers to a lookup
+    fn add_peers_to_lookup_and_ancestors(
+        &mut self,
+        lookup_id: SingleLookupId,
+        peers: &[PeerId],
+        cx: &mut SyncNetworkContext<T>,
+    ) -> Result<(), String> {
+        let lookup = self
+            .single_block_lookups
+            .get_mut(&lookup_id)
+            .ok_or(format!("Unknown lookup for id {lookup_id}"))?;
+
+        for peer in peers {
+            if lookup.add_peer(*peer) {
+                debug!(self.log, "Adding peer to existing single block lookup";
+                    "block_root" => ?lookup.block_root(),
+                    "peer" => ?peer
+                );
+            }
+        }
+
+        let id = lookup.id;
+        let awaiting_parent = lookup.awaiting_parent();
+        // After adding new peers a lookup may be able to make progress.
+        let result = lookup.continue_requests(cx);
```

Comment:
It's not clear to me why we try to progress to the lookup here, before we progress the parent. Wouldn't it make more sense to do it in reverse? Also this action feels quite out of place in this function although I'm not sure what's a better place for this.

Reply:
After https://github.com/sigp/lighthouse/pull/5839 lookups may have zero peers. I agree it won't be a common case that the ancestor lookup in a parent chain ends up with zero peers, pauses and then needs continuation from the code of this PR. I would be okay with dropping this new `continue_requests` statement, it's really an optimization and not strictly required